lerobot/common/robot_devices/robots/manipulator.py

```diff
@@ -445,134 +551,121 @@ def teleop_step(
-            raise ValueError(
-                f"You provided {follower_names=} but only the following arms are registered: "
-                f"{list(self.follower_arms)}"
+                "ManipulatorRobot is not connected. You need to run `robot.connect()`."
             )
 
+        # Get goal position of each follower arm by splitting the action vector
         from_idx = 0
         to_idx = 0
         follower_goal_pos = {}
-        for name in follower_names:
-            to_idx += len(self.follower_arms[name].motor_names)
-            this_action = action[from_idx:to_idx]
+        for name in self.follower_arms:
+            if name in self.follower_arms:
+                to_idx += len(self.follower_arms[name].motor_names)
+                follower_goal_pos[name] = action[from_idx:to_idx]
+                from_idx = to_idx
 
-            if self.config.max_relative_target is not None:
-                if not isinstance(self.config.max_relative_target, list):
-                    max_relative_target = [self.config.max_relative_target for _ in range(from_idx, to_idx)]
-                max_relative_target = torch.tensor(self.config.max_relative_target)
-                # Cap relative action target magnitude for safety.
-                current_pos = torch.tensor(self.follower_arms[name].read("Present_Position"))
-                diff = this_action - current_pos
-                safe_diff = torch.minimum(diff, max_relative_target)
-                safe_diff = torch.maximum(safe_diff, -max_relative_target)
-                safe_action = current_pos + safe_diff
-                if not torch.allclose(safe_action, this_action):
-                    logging.warning(
-                        "Relative action magnitude had to be clamped to be safe.\n"
-                        f"  requested relative action target: {diff}\n"
-                        f"    clamped relative action target: {safe_diff}"
-                    )
-                follower_goal_pos[name] = safe_action.numpy()
-            else:
-                follower_goal_pos[name] = this_action.numpy()
+        # Send goal position to each follower
+        for name in self.follower_arms:
+            goal_pos = follower_goal_pos[name]
 
-            from_idx = to_idx
+            # Cap goal position when too far away from present position.
+            # Slower fps expected due to reading from the follower.
+            if self.config.max_relative_target is not None:
+                present_pos = self.follower_arms[name].read("Present_Position")
+                present_pos = torch.from_numpy(present_pos)
+                goal_pos = ensure_safe_goal_position(goal_pos, present_pos, self.config.max_relative_target)
 
-        for name in self.follower_arms:
-            self.follower_arms[name].write("Goal_Position", follower_goal_pos[name].astype(np.int32))
+            goal_pos = goal_pos.numpy().astype(np.int32)
+            self.follower_arms[name].write("Goal_Position", goal_pos)
```

Comment:
Redesign of `max_relative_target` logic
cc @alexander-soare

Reply:
@Cadene could you also return the action that was executed? The function is returning nothing right now so it's an easy add, and I found this useful in RL + some other debugging things.